src/main/java/com/xpanxion/java/springboot/da1/demo/controller/student5/BookController5.java

```diff
@@ -26,30 +31,57 @@ public Book getBook(@PathVariable("id") Integer id) {
         var bookList = dataAccess.getBook(id);
         return bookList.get(0);
     }
+
+    @PostMapping("student5/api/v1/bookstores/{bookstoreId}/books")
+    public void insertBook(@PathVariable("bookstoreId") Integer bookstoreId, @RequestBody Book book) {
+        var bookId = dataAccess.insertBook(book);
+        dataAccess.insertBookIntoInventory(bookstoreId,bookId,1);
+    }
 }
 
 @Service
-class DataAccess {
+class BookDataAccess {
 
     //
     // Data Members
     //
 
     @Autowired
     private JdbcTemplate jdbcTemplate;
-    private final String SELECT_BOOK = "select * from book where book_id = ?";
+    @Autowired
+    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+
+    private final String SELECT_BOOK = "SELECT * FROM book WHERE book_id = ?";
+    private final String INSERT_BOOK = "INSERT INTO book (title, isbn, price) VALUES (:title, :isbn, :price)";
+    private final String INSERT_BOOK_INTO_INVENTORY = "INSERT INTO bookstore_book (bookstore_id, book_id, quantity) VALUES (:bookstore_id, :book_id, :quantity)";
 
     //
     // Constructors
     //
 
-    public DataAccess() {
+    public BookDataAccess() {
     }
 
     //
     // Methods
     //
 
+    public int insertBook(Book book) {
+        final KeyHolder keyId = new GeneratedKeyHolder();
+        SqlParameterSource params = new MapSqlParameterSource("title", book.getTitle())
+            .addValue("isbn", book.getIsbn())
+            .addValue("price", book.getPrice());
+        namedParameterJdbcTemplate.update(INSERT_BOOK, params, keyId);
+        return keyId.getKey().intValue();
+    }
+
+    public void insertBookIntoInventory(Integer bookstoreId, Integer bookId, Integer quantity) {
+        Map<String, Object> params = new HashMap<>();
```

Comment:
In general, we want to avoid using `Object` as a value because an object can be anything and we should try to narrow the scope of things whenever we can.  That in mind, let's replace this with `MapSqlParameterSource`.